reactor-core/src/main/java/reactor/core/Scannable.java

```diff
@@ -380,28 +395,77 @@ default String name() {
 	}
 
 	/**
-	 * Check this {@link Scannable}e and its {@link #parents()} for a name an return the
-	 * first one that is reachable.
-	 *
-	 * @return the name of the first parent that has one defined (including this scannable)
+	 * @return the default operatorName of this operator
 	 */
 	default String operatorName() {
-		String stripped = toString()
-				.replaceAll("Parallel|Flux|Mono|Publisher", "")
-				.replaceAll("Fuseable|Operator", "");
+		// /!\ this code is duplicated in `InnerConsumer#operatorName` in order to use simple class name instead of toString
+
+		/*
+		 * Strip an operator name of various prefixes and suffixes.
+		 * @param name the operator name, usually simpleClassName or fully-qualified classname.
+		 * @return the stripped operator name
+		 */
+		String name = toString();
+		if (name.contains("@") && name.contains("$")) {
+			name = name.substring(0, name.indexOf('$'));
+			name = name.substring(name.lastIndexOf('.') + 1);
+		}
+		String stripped = name
+				.replaceAll("Parallel|Flux|Mono|Publisher|Subscriber", "")
+				.replaceAll("Fuseable|Operator|Conditional", "");
 
 		if(stripped.length() > 0) {
 			return stripped.substring(0, 1).toLowerCase() + stripped.substring(1);
 		}
 		return stripped;
 	}
 
+	/**
+	 * List the operator names in the chain of operators (including the current element),
+	 * in their assembly order. This traverses the chain of operators both upstream and
+	 * downstream.
+	 * <ol>
+	 *     <li>if the current Scannable is a {@link Subscriber}, the chain can reach down to
+	 *     the final subscriber, provided it is {@link Scannable} (eg. lambda subscriber)</li>
+	 *     <li>if it is an operator the chain can reach up to the source, if it is a Reactor
+	 *     source (that is {@link Scannable}).</li>
+	 * </ol>
+	 *
+	 * @return a {@link Stream} of {@link #operatorName()} for each discovered operator in the chain
+	 */
+	default Stream<String> operatorChain() {
```

Comment:
I wonder if we should not keep a more neutral name (also for operatorName) in Scannable since it can apply to other dependency-like trace. 
WDYT about : dependencies() or stacktrace() ?